enhancements/kubelet/oci-image-volume.md

```diff
@@ -0,0 +1,178 @@
+  - https://issues.redhat.com/browse/OCPNODE-2929
+  - https://issues.redhat.com/browse/OCPNODE-2575
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+# OCI Image Volume
+
+## Summary
+
+In Kubernetes 1.33, support for OCI Image Volume went to Beta. This allows users to package files and share them among containers in a pod without including them in the main image, thereby reducing vulnerabilities and simplifying image creation.
+
+## Motivation
+
+Supporting OCI images directly as a VolumeSource allows Kubernetes to focus on OCI standards as well as allows it to store and distribute any content using OCI registries. This allows the project to grow into use cases which go beyond running particular images.
+
+### User Stories
+
+See [KEP-4639: User Stories](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#user-stories-optional)
+
+### Goals
+
+- Introduce a new VolumeSource type that allows mounting both OCI images and OCI artifacts.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing VolumeSource types.
+- This proposal does not address other use cases for OCI objects beyond directory sharing among containers in a pod.
+- Mounting thousands of images in a single pod.
+
+## Proposal
+
+1. Enable ImageVolume feature gate in 4.20
+
+    - This feature gate is disabled by default in Kubernetes 1.33, so we need to explicitly enable it for OCP
+
+
+
+2. Separate Graduation Approach between OCI Images and OCI Artifacts.
+
+    - **OCI Image Mount**: Promote to GA
+    - **OCI Artifact Mount**: Begin as DevPreview
+        - OCI Artifact functionality is less mature and requires further refinement
+        - Add a new CRI-O option to toggle OCI artifact mount support
+        - Disable OCI artifact mount by default
+
+### Workflow Description
+
+#### OCI Image
+
+Since OCI image mount will be GA in 4.20. No feature gate configuration is needed by users. Creating a pod with volumes.image will pull the image and mount in the container.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - name: test
+    image: registry.k8s.io/e2e-test-images/echoserver:2.3
+    volumeMounts:
+    - name: volume
+      mountPath: /volume
+  volumes:
+  - name: volume
+    image:
+      reference: quay.io/crio/artifact:v1
+      pullPolicy: IfNotPresent
+```
+
+#### OCI Artifact
+
+While OCI artifact is in dev preview, it fails to mount if a user specifies OCI artifact in reference. To enable the mount, users need to add this drop-in CRI-O configuration.
+
+```
+[crio.image]
+oci_artifact_mount_support = true
+```
+
+When it's enabled, users can mount OCI artifacts in the same way as OCI images. Once we feel comfortable with the CRI-O's implementation and make it GA, we'd remove this option. Instead of turning it off, we could introduce a new field in ContainerRuntimeConfig to toggle it.
```

Comment:
I would list the last sentece as an alternative, but I think we won't go that route (either we're comfortable with it or we aren't I think)